File: 2023-2024/sourcecode/mod1/lab2/12RecyclerViewBasicsJava/app/src/main/java/com/example/a12_recyclerviewbasics_java/WordListAdapter.java
package com.example.a12_recyclerviewbasics_java;

import android.view.LayoutInflater;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.a12_recyclerviewbasics_java.databinding.WordListItemBinding;

import java.util.ArrayList;

public class WordListAdapter extends RecyclerView.Adapter<WordListViewHolder> {

    private ArrayList<String> wordList;

    public WordListAdapter(ArrayList<String> wordList){
        this.wordList = wordList;
    }

    @NonNull
    @Override
    public WordListViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new WordListViewHolder(WordListItemBinding.inflate(
                LayoutInflater.from(parent.getContext()), parent, false
        ));
    }

    @Override
    public void onBindViewHolder(@NonNull WordListViewHolder holder, int position) {
        String currentItem = wordList.get(position);
        holder.bind(currentItem);
    }

    @Override
    public int getItemCount() {
        return wordList.size();
    }
}

File: 2023-2024/sourcecode/mod1/lab3/11RecyclerViewBasicsKotlin/app/src/main/java/com/example/a11_recyclerviewbasics_kotlin/WordListAdapter.kt
package com.example.a11_recyclerviewbasics_kotlin

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.example.a11_recyclerviewbasics_kotlin.databinding.WordListItemBinding

class WordListAdapter(private val wordList: MutableList<String>) :
    RecyclerView.Adapter<WordListViewHolder>() {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): WordListViewHolder {
        return WordListViewHolder(
            WordListItemBinding.inflate(
                LayoutInflater.from(parent.context), parent, false
            ))

    }

    override fun getItemCount(): Int = wordList.size

    override fun onBindViewHolder(holder: WordListViewHolder, position: Int) {
        val currentItem = wordList[position]
        holder.bind(currentItem)
    }

}